packages/modules-sdk/src/loaders/utils/load-internal.ts

```diff
@@ -0,0 +1,119 @@
+import { asFunction, asValue } from "awilix"
+import { createMedusaContainer } from "medusa-core-utils"
+import { trackInstallation } from "medusa-telemetry"
+import {
+  Constructor,
+  InternalModuleDeclaration,
+  Logger,
+  MedusaContainer,
+  ModuleExports,
+  ModuleResolution,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
+} from "../../types"
+
+export async function loadInternalModule(
+  container: MedusaContainer,
+  resolution: ModuleResolution,
+  logger: Logger
+): Promise<{ error?: Error } | void> {
+  const constainerName = resolution.definition.registrationName
+
+  const { scope, resources } =
+    resolution.moduleDeclaration as InternalModuleDeclaration
+
+  let loadedModule: ModuleExports
+  try {
+    loadedModule = (await import(resolution.resolutionPath as string)).default
+  } catch (error) {
+    return { error }
+  }
+
+  if (!loadedModule?.services || !loadedModule?.services.length) {
+    container.register({
+      [constainerName]: asValue(undefined),
+    })
+
+    return {
+      error: new Error(
+        "No service found in module. Make sure your module exports at least one service."
+      ),
+    }
```

Comment:
suggestion: Is it necessary to register undefined before returning the error? At that point if the module is there but there is no service, the app should crash so I would say that there is no need to register anything here.

If the app does not crash it means that it could start in the wrong state and you don't necessarily notice it right?

Reply:
If we don't set undefined the services calling `.resolve` will crash, and we use the message to display a warning when the server is loading.
The app doesn't start only in case the module is required

Reply:
Alright if we go that way then indeed it is needed